### PR TITLE
Add with-mocha example

### PR DESCRIPTION
**Pull request by msutkowski**
## Changes

- Adds a minified example from the [Mocha issue thread](https://github.com/mswjs/msw/issues/637)

## Example checklist

- [x] **I've followed the [Contribution guidelines](./CONTRIBUTING.md)**.
- [x] I've verified that all added tests pass by running `npm test`.
- [x] I've included only the necessary minimum setup to showcase the example.
- [x] I've added a `README.md` file to the root directory of my example.

### CODE REVIEW

**Comment by msutkowski** [comment]
I left this pretty close to the example issue and solution. I think this is general enough to show as an example for `mocha` users and explains the caveats _(no parallel mode support as of now)_

File: examples/with-mocha/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Karma
```
Comment (timdeschryver) [inline examples/with-mocha/README.md:1]:
```suggestion
# Mocha
```

File: examples/with-mocha/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "with-mocha",
+  "version": "0.0.0",
+  "scripts": {
+    "test": "cross-env NODE_ENV=test mocha --timeout 10000 src/**/*.test.js",
```
Comment (timdeschryver) [inline examples/with-mocha/package.json:5]: Could it be that the test files aren't included in the commit?

File: examples/with-mocha/README.md (hunk 1 above)
Comment (msutkowski) [reply examples/with-mocha/README.md:1]: Good catch.

File: examples/with-mocha/package.json (hunk 2 above)
Comment (msutkowski) [reply examples/with-mocha/package.json:5]: I sweat I've used git before... promise.

**Review by timdeschryver — DISMISSED** [review]

File: examples/with-mocha/src/Child.test.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import { render, screen, waitFor, cleanup } from '@testing-library/react'
+import { rest } from 'msw'
+import Child from './Child'
+import mswServer from '../mocks/server'
```
Comment (kettanaito) [inline examples/with-mocha/src/Child.test.js:5]: Do you think we could name it just `server`? I don't mind it being name differently, but we should stick to the same name throughout examples, which I think is `server`.

File: examples/with-mocha/src/Child.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          pageSize: 10,
+          pageNo: 0,
+          sortedColumn: 'createdBy',
+          sortedType: 'asc',
+        },
+      }),
+    })
+      .then(async (response) => {
+        const result = await response.json()
+        console.log('Fetch complete: ', result)
```
Comment (kettanaito) [inline examples/with-mocha/src/Child.js:23]: Looks like a left-over after debugging. Should we remove this `console.log`?

File: examples/with-mocha/test/registerModules.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  configFile: path.normalize(`${__dirname}/../babel.config.js`),
+  ignore: ['/node_modules'],
+})
+
+// Required per https://testing-library.com/docs/react-testing-library/setup#using-without-jest
+require('jsdom-global/register')
+
+// Support fetch in a node env. Requires the localstorage polyfill
+require('isomorphic-fetch')
+require('localstorage-polyfill')
```
Comment (kettanaito) [inline examples/with-mocha/test/registerModules.js:13]: Do we need the `localstorage-polyfill` if we use `whatwg-fetch` instead? 

I find it the less things our examples have, the more straightforward they are to follow.

File: examples/with-mocha/test/setupMocha.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+const { cleanup } = require('@testing-library/react')
+const server = require('../mocks/server')
+
+before(() => {
+  server.listen()
+  console.log(`msw server started!`)
```
Comment (kettanaito) [inline examples/with-mocha/test/setupMocha.js:6]: Nitpick: this `console.log` can be removed.

File: examples/with-mocha/test/setupMocha.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  server.printHandlers()
+})
+
+afterEach(() => {
+  // Reset any modified handlers and cleanup tests
+  server.resetHandlers()
+  cleanup()
```
Comment (kettanaito) [inline examples/with-mocha/test/setupMocha.js:13]: Doesn't RTL call `cleanup` automatically? Does that work only in Jest?

**Review by kettanaito — CHANGES_REQUESTED** [review]
Thanks for preparing this example, @msutkowski! I've left a few comments.

File: examples/with-mocha/test/registerModules.js (hunk 5 above)
Comment (msutkowski) [reply examples/with-mocha/test/registerModules.js:13]: @kettanaito localstorage would still need to be mocked or polyfilled here using whatwg-fetch. I can just include a mock if you think it's more transparent than including this polyfill or a mock lib, just let me know :)

File: examples/with-mocha/test/setupMocha.js (hunk 7 above)
Comment (msutkowski) [reply examples/with-mocha/test/setupMocha.js:13]: It should, but unfortunately, that's not the case here. If you add `console.log(screen.debug())` to the last test in `Child`, you'll see that it stacks components and the tests will fail.

File: examples/with-mocha/src/Child.js (hunk 4 above)
Comment (msutkowski) [reply examples/with-mocha/src/Child.js:23]: Yea, agreed.

**Comment by msutkowski** [comment]
@kettanaito Updated, but left the localstorage polyfill in place per the comment. If you want me to include a manual mock for this, just let me know :)

**Comment by kettanaito** [comment]
Thanks for all the work on this, @msutkowski! As I'm updating the examples, I think I will drop this one. The usage with mocha is rather straightforward, and mocha+react is a bit too complex to feature as a standalone example.

**Pull request closed without merging**